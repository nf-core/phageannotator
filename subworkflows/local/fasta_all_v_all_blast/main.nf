//
// Compare sequences by performing an all-v-all BLAST
//
include { BLAST_MAKEBLASTDB } from '../../../modules/nf-core/blast/makeblastdb/main'
include { BLAST_BLASTN      } from '../../../modules/nf-core/blast/blastn/main'

workflow FASTA_ALL_V_ALL_BLAST {

    take:
    fasta_gz    // [ [ meta ], fasta.gz ]   , assemblies/genomes (mandatory)

    main:
    ch_versions = Channel.empty()

    //
    // MODULE: Make BLASTN database
    //
    ch_blast_db = BLAST_MAKEBLASTDB ( fasta_gz ).db
    ch_versions = ch_versions.mix( BLAST_MAKEBLASTDB.out.versions )

    //
    // MODULE: Perform BLAST
    //
    ch_blast_txt = BLAST_BLASTN ( fasta_gz , ch_blast_db ).txt
    ch_versions = ch_versions = ch_versions.mix( BLAST_MAKEBLASTDB.out.versions )

    emit:
    blast_txt   = ch_blast_txt  // [ [ meta ], blast_output.tsv ]   , TSV file containing BLAST results
    versions    = ch_versions   // [ versions.yml ]

}
