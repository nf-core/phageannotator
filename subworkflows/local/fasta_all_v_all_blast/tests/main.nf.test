nextflow_workflow {

    name "Test Subworkflow: FASTA_ALL_V_ALL_BLAST"
    script "subworkflows/local/fasta_all_v_all_blast/main.nf"
    workflow "FASTA_ALL_V_ALL_BLAST"

    // Dependencies
    tag "BLAST_MAKEBLASTDB"
    tag "BLAST_BLASTN"


    test("fasta.gz") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
