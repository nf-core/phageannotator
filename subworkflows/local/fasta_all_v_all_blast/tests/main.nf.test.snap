{
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,aa3f583ccce0464e25ced18c56760404"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3a67b197cc055b730a0f1aa4d0cd31c0",
                    "versions.yml:md5,3a67b197cc055b730a0f1aa4d0cd31c0"
                ],
                "blast_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,aa3f583ccce0464e25ced18c56760404"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3a67b197cc055b730a0f1aa4d0cd31c0",
                    "versions.yml:md5,3a67b197cc055b730a0f1aa4d0cd31c0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T16:37:03.577392353"
    }
}
